pkg/api/protobuf/go/minder/v1/rule_types.go

```diff
@@ -67,6 +67,19 @@ func (r *RuleType) WithDefaultDisplayName() *RuleType {
 	return r
 }
 
+// WithDefaultShortFailureMessage sets the evaluation failure message if it is not set
+func (r *RuleType) WithDefaultShortFailureMessage() *RuleType {
+	if r == nil {
+		return nil
+	}
+
+	if r.ShortFailureMessage == "" {
+		r.ShortFailureMessage = "Rule evaluation failed"
```

Comment:
Since these will be shown as alerts, we should include the name in the failure message. For example, `"Rule %s evaluation failed", r.name`

Reply:
I've updated the message, thanks.

Reply:
Regarding the other message, it still works fine for me locally. I tested it with the following rule_type (just added the field to the existing secret_scanning rule_type):

```
---
version: v1
release_phase: beta
type: rule-type
name: secret_scanning
display_name: Enable secret scanning to detect hardcoded secrets
short_failure_message: Test message - Secret scanning is not enabled for the repository
severity:
  value: high
context:
  provider: github
description: |
  Verifies that secret scanning is enabled for a given repository.
  Note that this will will not work as expected for private repositories
  unless you have GitHub Advanced Security enabled. If you still want to use
  this rule because you have a mixture of private and public repositories,
  enable the `skip_private_repos` flag.
guidance: |
  Ensure that secret scanning is enabled for the repository.

  Secret scanning is a feature that scans repositories for secrets and
  alerts the repository owner when a secret is found. To enable this
  feature in GitHub, you must enable it in the repository settings.

  For more information, see [GitHub's
  documentation](https://docs.github.com/en/github/administering-a-repository/about-secret-scanning).
def:
  # Defines the section of the pipeline the rule will appear in.
  # This will affect the template used to render multiple parts
  # of the rule.
  in_entity: repository
  # Defines the schema for writing a rule with this rule being checked
  rule_schema:
    properties:
      enabled:
        type: boolean
        default: true
      skip_private_repos:
        type: boolean
        default: true
        description: |
          If true, this rule will be marked as skipped for private repositories
  # Defines the configuration for ingesting data relevant for the rule
  ingest:
    type: rest
    rest:
      # This is the path to the data source. Given that this will evaluate
      # for each repository in the organization, we use a template that
      # will be evaluated for each repository. The structure to use is the
      # protobuf structure for the entity that is being evaluated.
      endpoint: "/repos/{{.Entity.Owner}}/{{.Entity.Name}}"
      # This is the method to use to retrieve the data. It should already default to JSON
      parse: json
  # Defines the configuration for evaluating data ingested against the given profile
  eval:
    type: rego
    rego:
      type: deny-by-default
      def: |
        package minder

        import future.keywords.if

        default allow := false
        default skip := false

        allow if {
          input.profile.enabled
          input.ingested.security_and_analysis.secret_scanning.status == "enabled"
        }

        allow if {
          not input.profile.enabled
          input.ingested.security_and_analysis.secret_scanning.status == "disabled"
        }

        skip if {
          input.profile.skip_private_repos == true
          input.ingested.private == true
        }
  remediate:
    type: rest
    rest:
      method: PATCH
      endpoint: "/repos/{{.Entity.Owner}}/{{.Entity.Name}}"
      body: |
        {{- if .Profile.enabled }}
        { "security_and_analysis": {"secret_scanning": { "status": "enabled" } } }
        {{- else }}
        { "security_and_analysis": {"secret_scanning": { "status": "disabled" } } }
        {{- end }}
  # Defines the configuration for alerting on the rule
  alert:
    type: security_advisory
    security_advisory: {}

```

And I can see the record locally in the DB:

```
ae7d64a9-6bbd-45f0-99bf-7b7d2a743927,secret_scanning,,0fe3d233-c645-457c-896a-9b62f23dc6f0,"Verifies that secret scanning is enabled for a given repository.
Note that this will will not work as expected for private repositories
unless you have GitHub Advanced Security enabled. If you still want to use
this rule because you have a mixture of private and public repositories,
enable the `skip_private_repos` flag.
","Ensure that secret scanning is enabled for the repository.

Secret scanning is a feature that scans repositories for secrets and
alerts the repository owner when a secret is found. To enable this
feature in GitHub, you must enable it in the repository settings.

For more information, see [GitHub's
documentation](https://docs.github.com/en/github/administering-a-repository/about-secret-scanning).
","{""eval"": {""rego"": {""def"": ""package minder\n\nimport future.keywords.if\n\ndefault allow := false\ndefault skip := false\n\nallow if {\n  input.profile.enabled\n  input.ingested.security_and_analysis.secret_scanning.status == \""enabled\""\n}\n\nallow if {\n  not input.profile.enabled\n  input.ingested.security_and_analysis.secret_scanning.status == \""disabled\""\n}\n\nskip if {\n  input.profile.skip_private_repos == true\n  input.ingested.private == true\n}\n"", ""type"": ""deny-by-default""}, ""type"": ""rego""}, ""alert"": {""type"": ""security_advisory"", ""securityAdvisory"": {}}, ""ingest"": {""rest"": {""parse"": ""json"", ""endpoint"": ""/repos/{{.Entity.Owner}}/{{.Entity.Name}}""}, ""type"": ""rest""}, ""inEntity"": ""repository"", ""remediate"": {""rest"": {""body"": ""{{- if .Profile.enabled }}\n{ \""security_and_analysis\"": {\""secret_scanning\"": { \""status\"": \""enabled\"" } } }\n{{- else }}\n{ \""security_and_analysis\"": {\""secret_scanning\"": { \""status\"": \""disabled\"" } } }\n{{- end }}\n"", ""method"": ""PATCH"", ""endpoint"": ""/repos/{{.Entity.Owner}}/{{.Entity.Name}}""}, ""type"": ""rest""}, ""ruleSchema"": {""properties"": {""enabled"": {""type"": ""boolean"", ""default"": true}, ""skip_private_repos"": {""type"": ""boolean"", ""default"": true, ""description"": ""If true, this rule will be marked as skipped for private repositories\n""}}}}",2024-09-12 14:07:25.160724,2024-09-12 14:07:25.160724,high,,,Enable secret scanning to detect hardcoded secrets,beta,Test message - Secret scanning is not enabled for the repository

```

![image](https://github.com/user-attachments/assets/b6d46987-d8e5-43e8-b9d6-737172337a12)